tests/parser.js

```diff
@@ -1,56 +1,157 @@
 import test from 'ava';
 import parse from '../index';
 
-const SPACE = { word: ' ' };
-
 test('parsing', t => {
 
   t.deepEqual([], parse(''), 'empty input');
-  t.deepEqual([{ word: 'hi' }], parse('hi'), 'simple word');
-  t.deepEqual([{ variable: 'foo' }], parse('$foo'), 'simple var');
-
-  t.deepEqual([
-    { word: 'oh' },
-    SPACE,
-    { word: 'hi' },
-    SPACE,
-    { value: 'thing is quoted here' },
-    SPACE,
-    { hole: true },
-    SPACE,
-    { value: 1.0 },
-    SPACE,
-    { variable: 'var' }
-  ], parse('oh hi "thing is quoted here" _ 1.0 $var'));
+  t.deepEqual([ word('hi')], parse('hi'), 'simple word');
+  t.deepEqual([ variable('foo') ], parse('$foo'), 'simple var');
+
+  t.deepEqual(
+    ts()
+    .w_('oh')
+    .w_('hi')
+    .v_('thing is quoted here')
+    .h_()
+    .v_(1.0)
+    .variable('var').done(),
+    parse('oh hi "thing is quoted here" _ 1.0 $var')
+  );
+
+});
+
+test('whitespace', t => {
+  t.deepEqual(
+    ts().w_('hi').w('you').done(),
+    parse('hi    you'),
+    'white space gets collapsed'
+  );
+});
+
+test('words', t => {
+  // this exists to document *how* things are being parsed
+  // for documentation purposes
+  t.deepEqual(ts().w(',,,,').done(), parse(',,,,'));
+  t.deepEqual(ts().w('a,y').done(), parse('a,y'));
+  t.deepEqual(ts().w('a').h().done(), parse('a_'));
+  t.deepEqual(
+    ts().w('(').variable('a').w_(',').variable('b').w(')').done(),
+    parse('($a, $b)')
+  );
+  t.deepEqual(ts().w_('hi.').w('you').done(), parse('hi. you'));
+  t.deepEqual(ts().w('hi').v(0.1).done(), parse('hi.1'));
+  t.deepEqual(ts().w('w').v(1).done(), parse('w1'));
+  t.deepEqual(ts().w('w').v('').done(), parse('w""'));
+  t.deepEqual(ts().id('').done(), parse('#'));
+});
 
+test('FIXME quotes', t => {
+  // these are weird/ambiguous grammar rules that
```

Comment:
Having an open quote without a closing one should be an error in future versions.

Reply:
I wonder if test supports the semantic "We expect this to error, but test that it does not"
FIXME is a good way to mention it for now.